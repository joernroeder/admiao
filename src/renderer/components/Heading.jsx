//import styled from 'styled-components';
import styled from '@emotion/styled'
import { color, space } from 'styled-system'

import {
	verticalRhythm,
	verticalRhythmProps,
} from '../utils/styled-system-rhythm'

const ChapterHeading = styled.h1`
	${color};
	${space};
	${verticalRhythm};
	font-family: 'SpaceGrotesk', sans-serif;
	letter-spacing: -0.02em;
	text-rendering: geometricPrecision;
	-webkit-font-smoothing: antialiased;
	font-variant-ligatures: common-ligatures;
`

ChapterHeading.defaultProps = {
	...verticalRhythmProps([2, 2, 3]),
	color: 'black',
	my: 0,
}

export default ChapterHeading
