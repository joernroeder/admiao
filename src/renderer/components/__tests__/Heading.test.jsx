import React from 'react'

import { render } from '@testing-library/react'

import Heading from '../Heading'

describe('Heading component', () => {
	// todo test media queries: -> font-sizes and line-height
	// todo test vertical rhythm
	it('should render correctly', () => {
		const { container } = render(<Heading>Hello Jest</Heading>)
		const heading = container.firstChild

		expect(heading).toMatchSnapshot()
	})
})
