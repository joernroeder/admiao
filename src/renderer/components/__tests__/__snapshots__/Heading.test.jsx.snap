// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should render correctly 1`] = `
.emotion-0 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  line-height: 2;
  -webkit-transform: 2px;
  -ms-transform: 2px;
  transform: 2px;
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 20px;
    line-height: 3;
    -webkit-transform: 3px;
    -ms-transform: 3px;
    transform: 3px;
  }
}

<h1
  class="emotion-0"
  color="black"
  font-size="2,2,3"
>
  Hello Jest
</h1>
`;
